netket/graph/lattice.py

```diff
@@ -236,7 +151,8 @@ def __init__(
         atoms_coord: Optional[_np.ndarray] = None,
         distance_atol: float = 1e-5,
         point_group: Optional[PointGroup] = None,
-        max_neighbor_order: int = 1,
+        max_neighbor_order: Optional[int] = None,
+        custom_edges: Optional[Sequence] = None,
```

Comment:
In the docstring you say this should be a list of tuples... 

So i think the type could be improved to `Optional[Sequence[CustomEdgeT]]`
and define `CustomEdgeT = Union[Tuple[int, int, Array], Tuple[int,int,Vec,int]]` 

just for readability